pulp_container/app/registry_api.py

```diff
@@ -992,6 +963,8 @@ def put(self, request, path, pk=None):
                 objs=thru, ignore_conflicts=True, batch_size=1000
             )
 
+            add_content_units += [str(blob.pk) for blob in blobs_qs] + [str(config_blob.pk)]
+
         # a manifest cannot tagged by its digest - an identifier specified in the 'pk' parameter
         if not pk.startswith("sha256:"):
```

Comment:
I know that we agreed to first nail down the pending blobs, however, if i am not mistaken on the sequence of the manifest list API calls, it pushes children manifests by digest and not by tag - this way you could distinguish whether this PUT call is all about tagged image manifest or image manifest that is meant to be part of a list. Then you could add those to pending manifests and then clear out pending blob and manifests before creating the add_remove task.
Give it a try in a separate commit?

Reply:
I am doing it in one commit to reduce the number of separate migrations. Also, it is not that complex to implement and review it in one go.